webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/reports/ComputeNodeStatusReportService.scala

```diff
@@ -241,6 +279,41 @@ class ComputeNodeStatusReportServiceImpl(
     })
   }
 
+  /*
+   * Filter out reports that are marked as "NoReportInInterval" but have a policy to keep them longer
+   */
```

Comment:
this is the main logic that decides what needs to be done for reports that are in `NoRepertInInterval`